go/libcmdline/cmdline.go

```diff
@@ -613,7 +628,23 @@ func (p *CommandLine) PopulateApp(addHelp bool, extraFlags []cli.Flag) {
 		},
 		cli.StringFlag{
 			Name:  "proxy",
-			Usage: "Specify an HTTP(s) proxy to ship all Web requests over.",
+			Usage: "Specify a proxy to ship all Web requests over; Must be used with --proxy-type; Example: localhost:8080",
+		},
+		cli.StringFlag{
+			Name:  "proxy-type",
+			Usage: fmt.Sprintf("set the proxy type; One of: %s", func() string {
+				// This function retrieves the list of valid proxy types from the dict that tracks them
+				proxyTypes := ""
+				for k := range libkb.ProxyTypeStrToEnum {
```

Comment:
I'd make a helper here, so as not to make this thing even bigger.

Also, maybe pop the keys out into a slice and then call `strings.Join`, it might be slightly cleaner.

Reply:
Done